src/MergeContext.js

```diff
@@ -4,6 +4,43 @@ const Log = require('./Logger.js');
 const GH = require('./GitHubUtil.js');
 const Util = require('./Util.js');
 
+// Contains properties used for approval test status creation
+class Approval {
+
+    constructor(description, state, delayMs) {
+        assert(description);
+        assert(state);
+        this.description = description;
+        this.state = state;
+        // If waiting for a timeout to merge the staged commit: > 0
+        // If ready to merge the staged commit now: == 0
+        // Otherwise (e.g., failed description, negative votes, or review requested): null
+        this.delayMs = delayMs;
+    }
+
+    static GrantAfterTimeout(description, delayMs) {
+        assert(delayMs > 0);
+        return new Approval(description, "pending", delayMs);
+    }
+
+    static GrantNow(description) {
+        return new Approval(description, "success", 0);
+    }
+
+    static Block(description) {
+        return new Approval(description, "pending", null);
+    }
+
+    matchesGitHubStatusCheck(approvalStatus) {
+        assert(approvalStatus);
+        return approvalStatus.state === this.state && approvalStatus.description === this.description;
+    }
+
+    blocked() { return this.delayMs === null; }
```

Comment:
With the other changes discussed elsewhere applied, this may become something like return this.state === "failure" (or whatever the "red" state name is).

* blocked(): Cannot be merged unless GitHub state changes. No delay. Red status -- somebody needs to fix something.
* suspended(): Cannot be merged unless GitHub state changes. No delay. Yellow status -- somebody needs to add something.
* otherwise: Can be merged:
  * either immediately (i.e., zero delay()) (green status) or
  * after a positive delay() (yellow status).

I do not know whether the high-level Approval users need to know the difference between blocked() and suspended(). If they do not, then the two methods can be replaced with a single `!granted()` check. The Approval-creating code does know the difference, of course, so there will be two different static methods: Block() and Suspend().